backends-clickhouse/src/main/scala/org/apache/spark/sql/execution/joins/ClickHouseBuildSideRelation.scala

```diff
@@ -45,14 +45,17 @@ case class ClickHouseBuildSideRelation(
       broadCastContext: BroadCastHashJoinContext): ClickHouseBuildSideRelation = this
 
   private var hashTableData: Long = 0L
+  private var hashTableName: String = ""
+
   def buildHashTable(
       broadCastContext: BroadCastHashJoinContext): (Long, ClickHouseBuildSideRelation) =
     synchronized {
-      if (hashTableData == 0) {
+      if (hashTableData == 0 || hashTableName != broadCastContext.buildHashTableId) {
```

Comment:
use equals, not `!=`

Reply:
In scala , it seems that '!=' is better than equals.